src/test/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutesV1Spec.scala

```diff
@@ -38,6 +38,32 @@ class UserRoutesV1Spec extends UserRoutesSpecHelper{
     }
   }
 
+  it should "create a user if ToS is enabled and the user specifies the ToS parameter correctly" in withTosEnabledRoutes { samRoutes =>
+    Post("/register/user/v1?tos=app.terra.bio/#terms-of-service") ~> samRoutes.route ~> check {
```

Comment:
this test (and the v2 equivalent) are failing because `app.terra.bio/#terms-of-service` is not actually an allowed URL parameter, since it contains a forward slash. we'll need to either URL encode this parameter or change the URL that we accept. the reason i didn't jump right on URL encoding this was because that would also force the users to URL encode this parameter, and I'm not sure that we want to do that.

as-is, `/register/user/v1?tos=app.terra.bio/#terms-of-service` gets read as `/register/user/v1?tos=app.terra.bio`

Reply:
hm, good point! Going back to the original intent, the goal here is to prove _informed_ consent of the TOS … what can we ask the user to pass in that proves they know where to read the TOS? Some options I can think of:
* include the url `app.terra.bio/#terms-of-service` as-is, unencoded, in a POST body
* include the url, encoded, as a query param
* include two separate query params, something like "?toshost=app.terra.bio&toslocation=terms-of-service"

lastly - this may not be possible - use `/register/user/v1?tos=app.terra.bio#terms-of-service`. Note carefully that this omits the slash between "bio" and "terms-of-service". This relies on two things that need testing:
1. if a user enters `app.terra.bio#terms-of-service` in their browser, does it properly redirect to `https://app.terra.bio/#terms-of-service`? It does in my Chrome, but this may not be universal.
2. Can akka even see/retrieve the "#terms-of-service" portion of that URL? Technically it should be interpreted as a URL fragment and should not even be sent from the browser to the server, so akka may not have access to it at all.

Reply:
one more thought, maybe overkill: what if we stood up a new subdomain/webpage at something like "tos.terra.bio" and asked users to submit that value as the query param?